src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt

```diff
@@ -87,6 +93,18 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
         return null
     }
 
+    private fun lacksFreeMemory(): Boolean {
+        val runtime = Runtime.getRuntime()
+        val maxMemory = runtime.maxMemory()
+        if (maxMemory == Long.MAX_VALUE && !showedUnboundedMaxMemoryWarning) { // unbounded max memory
+            showedUnboundedMaxMemoryWarning = true
+            reporter.logWarning("Unbounded maximum JVM memory detected. This can potentially lead to OutOfMemoryError or " +
```

Comment:
This warning does not seem useful (no real use-cases where it helps), let's remove it for code simplicity